en/Ciphers/rsa_cipher.md

```diff
@@ -0,0 +1,162 @@
+# RSA Cryptography Algorithm
+
+## About RSA Algorithm
+RSA (Rivest–Shamir–Adleman) is one of the first public-key cryptosystems and is widely used for secure data transmission. In such a cryptosystem, the encryption key is public and distinct from the decryption key which is kept secret (private).
+
+## Algorithm
+
+1. Select 2 Prime Numbers - **p & q**
+2. Calculate `n` as $$n = p * q$$
+3. Calculate Euler's Totient Function of n, $$φ(n) = (p-1) * (q-1)$$
+4. Select PUBLIC KEY - $e$ , such that $e$ and $φ(n)$ are Co-primes, i.e, $$\gcd(e , φ(n))=1$$
+5. Calculate PRIVATE KEY, $d$ such that $$(d * e) \mod φ(n) = 1$$
+
+## Public and Private Keys
+
+1. The **Public  key** is `{ e , n }` which is known to all in the network.
+2. The **Private key** is `{ d , n }` which is known ONLY to the User to whom message is to be sent.
+
+## Encryption & Decryption
+
+#### Encryption Algorithm
+
+The Cipher Text, C is generated from the plaintext, M using the public key, e as:
+
+$$C = M^e \mod n$$
+
+#### Decryption Algorithm
+
+The Plain Text, M is generated from the ciphertext, C using the private key, d as:
+
+$$M = C^d \mod n$$
+
+<!-- ![RSA-block-diagram%20%281%29.png](https://github.com/TanmoySG/Cryptography-Algorithm/blob/master/RSA-Algorithm/diagrams/RSA-block-diagram-image.png) -->
+
+## Encryption & Decryption Mechanism
+
+The Encryption and Decryption mechanishm block diagram with sample message encryption.
+
+![RSA-Encryption-Decryption-block-diagram.png](https://github.com/TanmoySG/Cryptography-Algorithm/blob/master/RSA-Algorithm/diagrams/RSA-Encryption-Decryption-block-diagram.jpg)
+
+### Explanation
+
+The explanation for the above example,
+
+![RSA-Example-maths-only-diagram.png](https://github.com/TanmoySG/Cryptography-Algorithm/blob/master/RSA-Algorithm/diagrams/RSA-Example-maths-only-diagram.png)
+
+## Simple Implementation of RSA using Python
+
+```python
+from sympy import *
+import math 
+
+#Generate p and q
```

Comment:
Please add a space after the `#` (for all comments)

Reply:
Done